peer/test_utils.go

```diff
@@ -724,3 +725,49 @@ func createTestPeer(t *testing.T) *peerTestCtx {
 		chanStatusMgr: chanStatusMgr,
 	}
 }
+
+// startPeer invokes the `Start` method on the specified peer and handles any
+// initial startup messages for testing.
+func startPeer(t *testing.T, mockConn *mockMessageConn, peer *Brontide,
+	done chan struct{}) {
```

Comment:
ah, what i meant is that this can be done in this `startPeer` function. No need to pass it in from outside. 

But otherwise - yes this idea is correct 🎉

Reply:
I did it this way because of this:
https://github.com/lightningnetwork/lnd/pull/8631#discussion_r1593969436

Reply:
you could just do that check in this func too though (since it is part of the peer start process. Can then just have a `requirePeerStorage` boolean or something passed in here.

Reply:
https://github.com/lightningnetwork/lnd/pull/8490#discussion_r1591457646

Reply:
alright - how about then letting `startPeer` return a done channel rather than requiring that we pass one in? 

Passing in a channel is usually a way for a caller to signal something to the function (like, pls exit when i close this channel). But if we are purely using this channel to signal to the caller that we have exited, then we can create the function inside